jazelly/FinetuneLLMs
# Dockerize frontend

### Purpose
This is the initial stage of a series of DevOps PRs that helps us on streamlining the building process and reducing discrepancies among our development environments.

Changes contain the following

1. A `dockerfile` located in `frontend/` that describes basic build steps of our React frontend
2. A `docker-compose.yml` where we configured our containers
3. A `web/` directory where nginx configuration located
4. Migrated `.env.example.sh` into `env.dist` and created `init.sh` in root directory to streamline `.env` creation

### Functional test steps
1. Ensure Docker desktop is installed if you are on Windows (Optional)
2. Run `. init.sh; docker compose build; docker compose up -d` in root directory
3. Two containers `web` and `frontend` should be up in a minute
4. When those containers are ready, access http://localhost:3000/ and you should see the frontend landing page
![image](https://github.com/jazelly/FinetuneLLMs/assets/172100975/5eb73136-cdbe-4c81-9d2f-3c63d0c6eafa)
